plugins/destination/s3/client/client.go

```diff
@@ -80,3 +83,17 @@ func New(ctx context.Context, logger zerolog.Logger, spec specs.Destination) (de
 func (*Client) Close(ctx context.Context) error {
 	return nil
 }
+
+func (*Client) ResolveEndpoint(service, region string, options ...any) (aws.Endpoint, error) {
+	endpoint := os.Getenv("AWS_S3_ENDPOINT")
+	if endpoint == "" || service != s3.ServiceID {
+		return aws.Endpoint{}, &aws.EndpointNotFoundError{}
```

Comment:
- It will fallback to original implementation as documented in the SDK doc which was linked in the PR request.
- Since this is an S3 plugin I don't think having S3 in the method name is criticial.